src/Http/Http/src/Internal/DefaultConnectionInfo.cs

```diff
@@ -15,6 +16,8 @@ internal sealed class DefaultConnectionInfo : ConnectionInfo
         // Lambdas hoisted to static readonly fields to improve inlining https://github.com/dotnet/roslyn/issues/13624
         private readonly static Func<IFeatureCollection, IHttpConnectionFeature> _newHttpConnectionFeature = f => new HttpConnectionFeature();
         private readonly static Func<IFeatureCollection, ITlsConnectionFeature> _newTlsConnectionFeature = f => new TlsConnectionFeature();
+        // Should there be a default IConnectionLifetimeNotificationFeature which does nothing!?
```

Comment:
I don't think it should do nothing. Let's throw a clear error instead of potentially returning null.

Reply:
Or set the header?